### PR TITLE
Update feature demo examples

### PR DESCRIPTION
**Pull request by abuccts**
Update feature demo examples.

### CODE REVIEW

File: example/feature/README.md [hunk 1]
```diff
@@ -110,15 +110,15 @@ Within one VC, a high-priority job can preempt low-priority jobs.
...
 1. Use [hived-config-3](file/hived-config-3.yaml).
-2. Submit [itc-intra-imd-preempt-test](file/itc-intra-imd-preempt-test.yaml), which requests for 4 M60 GPUs for VC1 with test (0) priority.
-3. Submit [itc-intra-imd-preempt-prod](file/itc-intra-imd-preempt-prod.yaml), which also requests for 4 M60 GPUs for VC1 with prod (100) priority. The job will preempt the test job immediately, so the test job is retried and waiting for resource.
+2. Submit [itc-intra-imd-preempt-test](file/itc-intra-imd-preempt-test.yaml), which requests for 4 M60 GPUs for vc1 with test (0) priority.
+3. Submit [itc-intra-imd-preempt-prod](file/itc-intra-imd-preempt-prod.yaml), which also requests for 4 M60 GPUs for vc1 with prod (100) priority. The job will preempt the test job immediately, so the test job is retried and waiting for resource.
    <img src="file/itc-intra-imd-preempt-test.png" width="900"/>
    <img src="file/itc-intra-imd-preempt-prod.png" width="900"/>
 
 #### Lazy Preemption
 1. Use [hived-config-3](file/hived-config-3.yaml).
```
Comment (yqwang-ms) [inline example/feature/README.md:119]: Will double check it

File: example/feature/README.md [hunk 2]
```diff
@@ -68,8 +68,8 @@ This is useful for jobs that cannot perform any useful work, such as making prog
    <img src="file/itc-gang4.png" width="900"/>
 
 #### TensorFlow Distributed Training
-1. Use [hived-config-1](file/hived-config-1.yaml).
-2. Submit job [itc-dtf](file/itc-dtf.yaml) to VC2, it will success.
+1. Use [hived-config-2](file/hived-config-2.yaml).
```
Comment (yqwang-ms) [inline example/feature/README.md:71]: Why change to hived-config-2

File: example/feature/README.md [hunk 3]
```diff
@@ -129,8 +129,8 @@ Within one VC, a high-priority job can preempt low-priority jobs.
 One VC's [Guaranteed Job](#Guaranteed-Job) can preempt other VCs' [Opportunistic Jobs](#Opportunistic-Job).
 
 ### Reproduce Steps
-1. Use [hived-config-2](file/hived-config-2.yaml).
-2. Submit [itc-inter-preempt-oppo](file/itc-inter-preempt-oppo.yaml), which requests for 2 * 4 K80 GPUs for VC1 with oppo (-1) priority.
+1. Use [hived-config-3](file/hived-config-3.yaml).
+2. Submit [itc-inter-preempt-oppo](file/itc-inter-preempt-oppo.yaml), which requests for 2 * 4 K80 GPUs for vc1 with oppo (-1) priority.
```
Comment (yqwang-ms) [inline example/feature/README.md:133]: Why change to hived-config-3

File: example/feature/README.md (hunk 2 above)
Comment (abuccts) [reply example/feature/README.md:71]: each vc in hived-config-1 has only one node, the job needs 1 node plus 1 gpu and cannot be scheduled using hived-config-1

File: example/feature/README.md (hunk 3 above)
Comment (abuccts) [reply example/feature/README.md:133]: same, hived-config-2 doesn't have enough quota for step 3

File: pkg/algorithm/cell_allocation.go [hunk 4]
```diff
@@ -235,7 +235,7 @@ func getUsablePhysicalCells(
 		return nil
 	}
 	// prioritize the cells with fewer opportunistic pods (to reduce preemption)
-	sort.SliceStable(candidates, func(i, j int) bool {
+	sort.SliceStable(usableCandidates, func(i, j int) bool {
 		return candidates[i].GetUsedLeafCellNumAtPriorities()[opportunisticPriority] <
 			candidates[j].GetUsedLeafCellNumAtPriorities()[opportunisticPriority]
 	})
```
Comment (yqwang-ms) [inline pkg/algorithm/cell_allocation.go:241]: Should be:

```
 sort.SliceStable(usableCandidates, func(i, j int) bool {
  return usableCandidates[i].GetUsedLeafCellNumAtPriorities()[opportunisticPriority] <
   usableCandidates[j].GetUsedLeafCellNumAtPriorities()[opportunisticPriority]
 })
```

**Comment by yqwang-ms** [comment]
> 

How about make ignoreK8sSuggestedNodes=true is the default setting for hived?

We can rename it to awareK8sSuggestedNodes, and default to false/

---
Refers to: example/feature/file/itc-reconfig-4.yaml:32 in b5dce4f. [](commit_id = b5dce4f74ba3e2f3d9e83afd5abe389384699fbf, deletion_comment = False)

**Comment by yqwang-ms** [comment]
> 

Will here overwrite the  defaults.Set(&podSchedulingSpec)
?

---
Refers to: pkg/internal/utils.go:245 in a54c3d8. [](commit_id = a54c3d8afeef6ff91b21dd7879529453594f49c8, deletion_comment = False)

**Comment by yqwang-ms** [comment]
> 

Still prefer to rename it to be awareK8sSuggestedNodes or make it a pointer, instead of depends on other pkg (github.com/creasty/defaults is not offical and very less stars)

---
In reply to: [675337957](https://github.com/microsoft/hivedscheduler/pull/30#issuecomment-675337957) [](ancestors = 675337957)

---
Refers to: pkg/internal/utils.go:245 in a54c3d8. [](commit_id = a54c3d8afeef6ff91b21dd7879529453594f49c8, deletion_comment = False)

**Comment by yqwang-ms** [comment]
> 

If we decide to make true as default. Maybe you need to revisit you already tested demos, as you may need to test again them under the "true" change.

---
In reply to: [674816970](https://github.com/microsoft/hivedscheduler/pull/30#issuecomment-674816970) [](ancestors = 674816970)

---
Refers to: example/feature/file/itc-reconfig-4.yaml:32 in e99765a. [](commit_id = e99765a00053c760a0a520bd06de3c31a4338342, deletion_comment = False)

**Comment by yqwang-ms** [comment]
> 

bad node demo can use default setting, i.e. ignoreK8sSuggestedNodes=true.

---
Refers to: example/feature/file/itc-badnode50.yaml:33 in 368ec2b. [](commit_id = 368ec2b01dbccbb957e1156cfaea74abcb1b560d, deletion_comment = False)

**Comment by yqwang-ms** [comment]
> ### Description

Remove?

---
Refers to: example/feature/README.md:224 in 058444f. [](commit_id = 058444f658a91e6b5b073b2e7efa6a8afc23e2a0, deletion_comment = False)

**Comment by yqwang-ms** [comment]
Search "Leverage-K8S-Default-Scheduler" and remove all

**Review by yqwang-ms — APPROVED** [review]

**Comment by zhypku** [comment]
> Search "Leverage-K8S-Default-Scheduler" and remove all

So are we abandoning this feature? :)

**Comment by yqwang-ms** [comment]
> > Search "Leverage-K8S-Default-Scheduler" and remove all
> 
> So are we abandoning this feature? :)

Temp disable it, as it may cause safety issue after reconfig (lazy preempted pod's k8s side pod priority does not change, so it is not in suggested nodes even during preemption phase, ORZ....)

**Pull request merged**